Add `swaggerUrl` to setup function, as alternative to `swaggerDoc`

skray opened the pull request:
Following up on #20, this PR adds the `swaggerUrl` option to the `setup` function.

This allows things like:

1. Allowing relative `$ref` paths if your swagger is split into multiple files
1. Showing the link to the swagger definition in the UI without having to specify the path in the URL query string
1. Checking your swagger against the online validator (if it's accessible to the open internet)

I wanted to get this up here to get feedback, but I still have to:

1. Get the tests running, I've tested the changes manually but I'm having some trouble getting the automated tests to pass, so I'll take another shot at that tomorrow
1. Update the README

scottie1984 commented:
Looks good. 👍 

1. I was having some problems with the tests as well when I upgraded to the latest version of swagger ui (I think some of the elements I am looking for in the tests have changed). Its on my todo list to fix but don't worry as long as you have tested manually. (if you can get them working then that would be great)

2. That would be great.

skray commented:
OK:

Fixed tests and added one for my changes, also added docs for the new param.

Let me know if I picked the right selectors for the test, those were the ids I saw when serving the test page.

Also, I had to bump up the timeout for the initial page load in the test significantly, it was taking up to ~25 seconds to load the page.

scottie1984 commented:
Should be in there in now. Thanks for your help.

skray commented:
Awesome, thanks for getting that in!
